File: easy/e70/golang/solution_test.go
package golang

import "testing"

func TestClimbStairs(t *testing.T) {
	input := []int{2, 3, 4, 5, 45}
	expectations := []int{2, 3, 5, 8, 1836311903}

	if len(input) != len(expectations) {
		t.Fatal("test is invalid")
	}

	for i := 0; i < len(input); i++ {
		result := climbStairs(input[i])

		if result != expectations[i] {
			t.Fatalf("\nbad result for %d\nexpected: %d\ngot: %d\n",
				input[i], expectations[i], result)
		}
	}
}

File: easy/e70/typescript/solution_test.ts
import { climbStairs } from "./solution"

function testClimbStairs(): string {
    let input1: number[] = [2, 3, 4, 5, 45];

    let expectations: number[] = [2, 3, 5, 8, 1836311903];

    if (input1.length != expectations.length) {
        return "test is invalid";
    }

    for (let i = 0; i < expectations.length; i++) {
        let result = climbStairs(input1[i]);

        let msg = `bad result for ${input1[i]}: got ${result} expected ${expectations[i]}`;

        if (result != expectations[i]) {
            return msg;
        }
    }

    return "PASS";
}

console.log(testClimbStairs());